src/main/java/mate/academy/spring/service/impl/UserServiceImpl.java

```diff
@@ -1,27 +1,26 @@
 package mate.academy.spring.service.impl;
 
 import java.util.Optional;
+import mate.academy.spring.config.SecurityConfig;
 import mate.academy.spring.dao.UserDao;
 import mate.academy.spring.model.User;
 import mate.academy.spring.service.UserService;
-import mate.academy.spring.util.HashUtil;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
-    private static final int SALT_LENGTH = 10;
     private final UserDao userDao;
+    private final SecurityConfig securityConfig;
 
-    public UserServiceImpl(UserDao userDao) {
+    public UserServiceImpl(UserDao userDao, SecurityConfig securityConfig) {
         this.userDao = userDao;
+        this.securityConfig = securityConfig;
     }
 
     @Override
     public User add(User user) {
-        String salt = HashUtil.getSalt(SALT_LENGTH);
-        String securePassword = HashUtil.generateSecurePassword(user.getPassword(), salt);
+        String securePassword = securityConfig.getEncoder().encode(user.getPassword());
```

Comment:
**12. Don’t create PasswordEncoder using new keyword. Let’s use dependency injection by the constructor.**

https://mate-academy.github.io/jv-program-common-mistakes/java-spring/security/jv-spring-security-checklist